tensorflow/python/keras/layers/wrappers.py

```diff
@@ -148,56 +150,73 @@ def _get_shape_tuple(self, init_tuple, tensor, start_idx, int_shape=None):
         of the output shape
 
     Returns:
-      The new int_shape with the first part from init_tuple
-      and the last part from either `int_shape` (if provided)
-      or `tensor.shape`, where every `None` is replaced by
-      the corresponding dimension from `tf.shape(tensor)`.
+      A tensor with the new int_shape with the first part from
+      init_tuple and the last part from either `int_shape`
+      (if provided) or `tensor.shape`, where every `None` is
+      replaced by the corresponding dimension from `tf.shape(tensor)`.
     """
     # replace all None in int_shape by K.shape
     if int_shape is None:
       int_shape = K.int_shape(tensor)[start_idx:]
     if not any(not s for s in int_shape):
-      return init_tuple + tuple(int_shape)
+      return convert_to_tensor(init_tuple + tuple(int_shape))
     shape = K.shape(tensor)
     int_shape = list(int_shape)
     for i, s in enumerate(int_shape):
       if not s:
         int_shape[i] = shape[start_idx + i]
-    return init_tuple + tuple(int_shape)
+    return convert_to_tensor(init_tuple + tuple(int_shape))
 
   def build(self, input_shape):
-    input_shape = tensor_shape.TensorShape(input_shape).as_list()
-    if len(input_shape) < 3:
+    input_shape = tf_utils.convert_shapes(input_shape, to_tuples=False)
+    input_dims = generic_utils.to_list(nest.map_structure(lambda x: x.ndims, input_shape))
```

Comment:
please use nest.flatten to replace generic_utils.to_list, which should be faster and more standard approach.